R/pr.R

```diff
@@ -58,38 +58,55 @@ pr_fetch <- function(number, owner = NULL) {
   check_branch_current("master")
   check_uncommitted_changes()
 
-  ui_done("Retrieving PR data")
+  ui_done("Retrieving data for PR #{number}")
   pr <- gh::gh("GET /repos/:owner/:repo/pulls/:number",
     owner = owner %||% github_owner(),
     repo = github_repo(),
     number = number
   )
 
-  ref <- pr$head$ref
-  user <- pr$head$user$login
-  if (user == github_owner()) {
-    user <- "origin"
-    branch <- ref
+  their_branch <- pr$head$ref
+  them <- pr$head$user$login
+  if (them == github_owner()) {
+    remote <- "origin"
+    our_branch <- their_branch
   } else {
-    branch <- paste0(user, "-", ref)
+    remote <- them
+    our_branch <- glue("{them}-{their_branch}")
   }
-  remote <- paste0(user, "/", ref)
 
-  if (!user %in% git2r::remotes(git_repo())) {
-    ui_done("Adding remote {ui_value(user)}")
-    git2r::remote_add(git_repo(), user, pr$head$repo$ssh_url)
+  if (!remote %in% git2r::remotes(git_repo())) {
+    ui_done("Adding remote {ui_value(remote)}")
+    git2r::remote_add(git_repo(), remote, pr$head$repo$ssh_url)
   }
 
-  if (!git_branch_exists(branch)) {
-    ui_done("Creating local branch {ui_value(branch)}")
-    git2r::fetch(git_repo(), user, refspec = ref, verbose = FALSE)
-    git_branch_create(branch, remote)
-    git_branch_track(branch, user, ref)
+  if (!git_branch_exists(our_branch)) {
+    their_refname <- git_remref(remote, their_branch)
+
+    ui_done("Creating local branch {ui_value(our_branch)}")
+    git2r::fetch(git_repo(), remote, refspec = their_branch, verbose = FALSE)
+    git_branch_create(our_branch, their_refname)
+
+    git_branch_track(our_branch, remote, their_branch)
+
+    ui_done("Caching URL for PR #{number} in config for branch {ui_value(our_branch)}")
```

Comment:
I'm not sure it's useful to expose this to the user. It seems like an internal detail?